### PR TITLE
chore: Replace logging with tokio::tracing

### PR DESCRIPTION
**Pull request by frondeus**

### CODE REVIEW

File: crates/api/src/config.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ pub struct Config {
 
     /// Kafka topic/AMQP queue on which API listens for notifications
     #[structopt(long, env)]
-    pub report_source: String,
+    pub report_source: Option<String>,
```
Comment (frondeus) [inline crates/api/src/config.rs:21]: Some changes (like this one) I made to fix CDL and be able to run the whole stack locally.

File: crates/api/src/schema/query.rs [hunk 2]
```diff
@@ -43,80 +44,95 @@ impl Schema {
...
-                    name: view.name,
-                    expression: view.jmespath,
+        let span = tracing::trace_span!("query_views", ?self.id);
+        async move {
+            let mut conn = context.connect_to_registry().await?;
+            let id = self.id.to_string();
+            let rpc_id = rpc::schema_registry::Id { id: id.clone() };
+
+            let views = conn
+                .get_all_views_of_schema(rpc_id.clone())
```
Comment (luke-biel) [inline crates/api/src/schema/query.rs:117]:
```suggestion
                .get_all_views_of_schema(rpc_id)
```

File: crates/api/src/schema/query.rs [hunk 3]
```diff
@@ -191,46 +226,56 @@ impl Query {
...
-            .collect::<Vec<CdlObject>>())
+        let span = tracing::trace_span!("query_objects", ?object_ids, ?schema_id);
+        async move {
+            let client = reqwest::Client::new();
+
+            let id_list = object_ids
+                .into_iter()
+                .map(|id| id.to_string())
+                .collect::<Vec<String>>()
+                .join(",");
```
Comment (luke-biel) [inline crates/api/src/schema/query.rs:237]: By relying on fmt::Display you can avoid allocation (this requires Itertools)

```
            let id_list = object_ids
                .iter()
                .join(",");
```

File: deployment/compose/docker-compose.cdl.grpc.postgres.yml [hunk 4]
```diff
@@ -24,7 +24,8 @@ services:
       POSTGRES_PORT: 5432
       POSTGRES_DBNAME: postgres
       POSTGRES_SCHEMA: cdl
-      REPORT_DESTINATION: "http://api:50104"
+      REPORT_ENDPOINT_URL: "http://api:50104"
```
Comment (luke-biel) [inline deployment/compose/docker-compose.cdl.grpc.postgres.yml:27]: Shouldn't this reflect the change to api port?

File: crates/command-service/src/args.rs [hunk 5]
```diff
@@ -51,7 +51,7 @@ pub struct CommunicationArgs {
     #[structopt(long, env)]
     pub amqp_consumer_tag: Option<String>,
     /// Port to listen on
-    #[structopt(long = "rpc-input-port", env = "RPC_PORT")]
+    #[structopt(long = "rpc-input-port", env = "GRPC_PORT")]
```
Comment (luke-biel) [inline crates/command-service/src/args.rs:54]: Just to be sure - this fixes the previous typo and is already reflected in docs/deployments?

**Review by luke-biel — CHANGES_REQUESTED** [review]
I'm aware that clones were already there, but since the API port needs changing, could you fix them?

Also - I believe `edge-registry` is in and should be included. I can do that if you are busy, just give me a notice.

File: crates/command-service/src/args.rs (hunk 5 above)
Comment (frondeus) [reply crates/command-service/src/args.rs:54]: Exactly.

File: deployment/compose/docker-compose.cdl.grpc.postgres.yml (hunk 4 above)
Comment (frondeus) [reply deployment/compose/docker-compose.cdl.grpc.postgres.yml:27]: It should in theory, but this is more a placeholder - API right now supports report notifications only from Kafka or AMQP.

**Comment by frondeus** [comment]
> I'm aware that clones were already there, but since the API port needs changing, could you fix them?
> 
> Also - I believe `edge-registry` is in and should be included. I can do that if you are busy, just give me a notice.

Done & done

**Review by luke-biel — APPROVED** [review]

**Comment by Kononnable** [comment]
- How does our logging currently look like? I mean we all knew old format (`path: message`), what is displayed after those changes?
- Do we have to set `RUST_LOG` like variable to enable logging or does it support this one?
- Does our tracing reporter support also `log` syntax?  Can we enable logs from rdkafka by changing environmental variables or do we have to add back `env_logger`?

**Comment by frondeus** [comment]
> * How does our logging currently look like? I mean we all knew old format (`path: message`), what is displayed after those changes?

An example

So when you use `tracing::instrument`, the message format looks like:
(`call site with arguments: path: message`) where call site can look like `Server::run{addr=0.0.0.0:50106}`.
Please keep in mind that these are structural traces, and while it might look inconvenient in the console output, we could use for example [OpenTelemetry](https://github.com/tokio-rs/tracing/tree/master/tracing-opentelemetry) to collect traces, or send them to for example GreyLog and filter them.

> * Do we have to set `RUST_LOG` like variable to enable logging or does it support this one?

You use `RUST_LOG` just like with `log` crate.

> * Does our tracing reporter support also `log` syntax?  Can we enable logs from rdkafka by changing environmental variables or do we have to add back `env_logger`?

While I didn't enable it we could use a compatibility layer to read `log::debug` from dependencies and treat them as traces.

**Comment by frondeus** [comment]
> While I didn't enable it we could use a compatibility layer to read `log::debug` from dependencies and treat them as traces.

One note - it is true what I wrote, however now I see that `tracing_subscriber` has one optional dependency:

> tracing-log: Enables better formatting for events emitted by log macros in the fmt subscriber. On by default.

Therefore it should print also logs next to traces. It won't, however, send logs to `GELF`/OpenTracing etc.

**Review by luke-biel — APPROVED** [review]

**Pull request merged**